apps/framework-cli/src/framework/schema.rs

```diff
@@ -280,53 +311,75 @@ fn field_to_column(f: &Field) -> Result<Column, ParsingError> {
     }
 }
 
-fn top_to_schema(t: &Top) -> Result<DataModel, ParsingError> {
-    match t {
-        Top::Model(m) => {
-            let schema_name = m.name().to_string();
-            let mut version = 1;
-
-            let attributes = m.attributes();
-
-            // Get the value of the version attribute in the ugliest way possible
-            let version_attribute = attributes.iter().find(|a| a.name() == "version");
-
-            if let Some(attribute) = version_attribute {
-                version = attribute
-                    .arguments
-                    .arguments
-                    .first()
-                    .map(|arg| {
-                        arg.value
-                            .as_numeric_value()
-                            .unwrap()
-                            .0
-                            .parse::<i8>()
-                            .unwrap()
-                    })
-                    .unwrap_or(version);
-            }
-
-            let columns: Result<Vec<Column>, ParsingError> =
-                m.iter_fields().map(|(_id, f)| field_to_column(f)).collect();
-
-            Ok(DataModel {
-                db_name: "local".to_string(),
-                columns: columns?,
-                name: schema_name,
-                version,
+fn top_to_datamodel(m: &Model, enums: &Vec<DataEnum>) -> Result<DataModel, ParsingError> {
+    let schema_name = m.name().to_string();
+    let mut version = 1;
+
+    let attributes = m.attributes();
+
+    // Get the value of the version attribute in the ugliest way possible
+    let version_attribute = attributes.iter().find(|a| a.name() == "version");
+
+    if let Some(attribute) = version_attribute {
+        version = attribute
+            .arguments
+            .arguments
+            .first()
+            .map(|arg| {
+                arg.value
+                    .as_numeric_value()
+                    .unwrap()
+                    .0
+                    .parse::<i8>()
+                    .unwrap()
             })
-        }
-        _ => Err(ParsingError::UnsupportedDataTypeError {
-            type_name: "we don't currently support anything other than models".to_string(),
-        }),
+            .unwrap_or(version);
     }
+
+    let columns: Result<Vec<Column>, ParsingError> = m
+        .iter_fields()
+        .map(|(_id, f)| field_to_column(f, enums))
+        .collect();
+
+    Ok(DataModel {
+        db_name: "local".to_string(),
+        columns: columns?,
+        name: schema_name,
+        version,
+    })
 }
 
-pub fn ast_mapper(ast: SchemaAst) -> Result<Vec<DataModel>, ParsingError> {
-    ast.iter_tops()
-        .map(|(_id, t)| top_to_schema(t))
-        .collect::<Result<Vec<DataModel>, ParsingError>>()
+fn top_to_enum(e: &Enum) -> DataEnum {
```

Comment:
Not a blocker - but please take the time to have fully spelled out names for variables. It makes it easier to read the code.

Suggestion here for the function signature:

```rust
fn convert_prisma_enum_to_moose_enum(prisma_enum: &Enum) -> DataEnum {
```

Reply:
I can live without the "convert".